### PR TITLE
Expand metaicl

### PR DESCRIPTION
**Pull request by jagnusson**
Adds support for QA, NLI, Paraphrase, and low resource evaluation task sets in MetaICL.

### zeroshot
Screen Shot 2022-08-08 at 8 11 52 PM

### fewshot
A quick check confirms results on `tweet_eval-stance_atheism` and `yelp_polarity` still reproduce their counterparts from the MetaICL repo. These are results using the non_class_to_class metaICL weights with 16 shot and a fewshot seed of 100.

#### using catwalk
```
metaicl::tweet_eval-stance_atheism      100     acc     0.25
metaicl::tweet_eval-stance_atheism      100     f1      0.17949944734573364
metaicl::tweet_eval-stance_atheism      100     precision       0.1979166716337204
metaicl::tweet_eval-stance_atheism      100     recall  0.21863798797130585

metaicl::yelp_polarity  100     acc     0.9570000171661377
metaicl::yelp_polarity  100     f1      0.9569902420043945
metaicl::yelp_polarity  100     precision       0.9569412469863892
metaicl::yelp_polarity  100     recall  0.9571332931518555
```

#### using MetaICL Repo
results for tweet_eval-stance_atheism
```   
07/21/2022 11:17:40 - INFO - __main__ - Macro-F1 of None over 1 target tasks: 17.9  
```

results for yelp_polarity
```
07/21/2022 11:22:23 - INFO - __main__ - Macro-F1 of None over 1 target tasks: 95.7
```

### CODE REVIEW

File: catwalk/tasks/metaicl.py [hunk 1]
```diff
@@ -87,17 +87,17 @@ def instance_as_rank_classification(
             fewshot_instances = []
         prefix = ""
         for fewshot_instance in fewshot_instances:
-            as_rc = self.instance_as_rank_classification(fewshot_instance)
+            as_rc = self.instance_as_rank_classification(fewshot_instance, continuation_seperator=continuation_seperator, example_seperator=example_seperator)
             if as_rc.correct_choice is None:
                 raise ValueError("Could not determine correct choice in ranked classification instance.")
             correct_choice = as_rc.choices[as_rc.correct_choice]
-            prefix += f"{correct_choice[0].strip()}{continuation_seperator}{correct_choice[1].strip()}{example_seperator}"
+            prefix += correct_choice[0] + correct_choice[1] + example_seperator
```
Comment (dirkgr) [inline catwalk/tasks/metaicl.py:94]: There is no separator between the choices?

**Review by dirkgr — APPROVED** [review]

File: catwalk/tasks/metaicl.py (hunk 1 above)
Comment (jagnusson) [reply catwalk/tasks/metaicl.py:94]: The separator is already included when the fewshot_instances go into `instance_as_rank_classification()`themselves. The old code would strip that and reapply the separator here but that was problematic for examples that have leading or trailing whitespace that needs to be kept to reproduce metrics.

Comment (dirkgr) [reply catwalk/tasks/metaicl.py:94]: That's fine. But out of curiosity, why does the white space still matter after tokenization?

Comment (jagnusson) [reply catwalk/tasks/metaicl.py:94]: Oh yes I had the same thought! It turns out that gpt2 tokenization has an explicit trailing space token (`220` I think). It has to use it when a space is at the end of an input and thus can't be attached to the next token.

**Pull request merged**